apps/smtp/src/modules/smtp/configuration/smtp-metadata-manager.ts

```diff
@@ -27,7 +40,19 @@ export class SmtpMetadataManager {
   > {
     this.logger.debug("Fetching SMTP config");
 
-    return fromPromise(this.metadataManager.get(this.metadataKey, this.saleorApiUrl), (e) => {
+    const timeoutedPromise = racePromise({
+      promise: this.metadataManager.get(this.metadataKey, this.saleorApiUrl),
+      timeout: this.pullConfigTimeout,
+      error: new SmtpMetadataManager.TimeoutExceededError("Timeout while fetching metadata"),
+    });
+
+    return fromPromise(timeoutedPromise, (e) => {
+      if (e instanceof SmtpMetadataManager.TimeoutExceededError) {
+        this.logger.debug("Fetching config exceeded timeout", { error: e });
+
+        return e;
+      }
+
       this.logger.debug("Failed to fetch config", { error: e });
```

Comment:
there is no visibility of DEBUG level logs because they are disabled on deployed env

Reply:
errors are handled by the useCase class, then it's logged into DD + sentry when it is an unhandled type of error. But since it's a wrapping error into error - will add this log to have better visibility